tests/Serializer/SerializerTest.php

```diff
@@ -96,6 +98,63 @@ public function testNull(bool $serializeAllObjects): void
         $this->assertNull($result);
     }
 
+    public function testRegisteredObjectSerializers(): void
+    {
+        $serializer = $this->createSerializer(new Options([
+            'class_serializers' => [
+                StubObject::class => $customSerializerCallback = static function (StubObject $object): array {
+                    return [
+                        'purpose' => $object->getPurpose(),
+                    ];
+                },
+            ],
+        ]));
+
+        $object = new StubObject();
+
+        $this->assertEquals([
+            'class' => \get_class($object),
+            'data' => $customSerializerCallback($object),
+        ], $this->invokeSerialization($serializer, $object));
+    }
+
+    public function testSerializableObject(): void
+    {
+        $serializer = $this->createSerializer();
+
+        $serializedValue = [
+            'testing' => 'value',
+        ];
+
+        $object = $this->createMock(SerializableInterface::class);
+        $object->method('toSentry')->willReturn($serializedValue);
+
+        $this->assertEquals([
+            'class' => \get_class($object),
+            'data' => $serializedValue,
+        ], $this->invokeSerialization($serializer, $object));
+    }
+
+    public function testSerializableThatReturnsNull(): void
+    {
+        $serializer = $this->createSerializer();
+
+        $object = $this->createMock(SerializableInterface::class);
+        $object->method('toSentry')->willReturn(null);
+
+        $this->assertEquals('Object ' . \get_class($object), $this->invokeSerialization($serializer, $object));
+    }
+
+    public function testSerializableObjectThatThrowsAnException(): void
+    {
+        $serializer = $this->createSerializer();
+
+        $object = $this->createMock(SerializableInterface::class);
+        $object->method('toSentry')->willThrowException(new \Exception('Doesn\'t matter what the exception is.'));
```

Comment:
Since this is a fluent interface, please split each call to a new line to ease review, debugging and readability